runtime/jvmti/suspendhelper.cpp

```diff
@@ -54,7 +62,11 @@ suspendThread(J9VMThread *currentThread, jthread thread, UDATA allowNull, UDATA
 						}
 					}
 					omrthread_monitor_exit(targetThread->publicFlagsMutex);
+
 					currentThread->javaVM->internalVMFunctions->internalEnterVMFromJNI(currentThread);
+
+					/* Release java.lang.Thread.lock */
+					objectMonitorExit(currentThread, threadLock);
```

Comment:
The local can't be re-used as the object may move while the thread is suspended.

Reply:
The lock object needs to be refetched from the vmThread.